fix a Scrollbar example crash

xu-baolin opened the pull request:
Fix a scrollbar example crash.
https://api.flutter.dev/flutter/material/Scrollbar-class.html#material.Scrollbar.1

TahaTesser commented:
Hi @xu-baolin 
I cannot reproduce any crash using the sample https://api.flutter.dev/flutter/material/Scrollbar-class.html#material.Scrollbar.2, on the latest stable or previous stable or master.

xu-baolin commented:
Did you have tried to move the scrollbar thumb?

________________________________
来自TT移动邮箱 / From TT mailbox


________________________________
发件人: Taha Tesser ***@***.***>
发送时间: 2023年06月29日 21:06
收件人: flutter/flutter ***@***.***>
抄送: 徐宝林(Ava) ***@***.***>; Mention ***@***.***>
主题: Re: [flutter/flutter] fix a Scrollbar example crash (PR #127925)


Hi @xu-baolin<https://github.com/xu-baolin>
I cannot reproduce any crash using the sample https://api.flutter.dev/flutter/material/Scrollbar-class.html#material.Scrollbar.2, on the latest stable or previous stable or master.

―
Reply to this email directly, view it on GitHub<https://github.com/flutter/flutter/pull/127925#issuecomment-1613153340>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOR66GBHBPEJYA7TVGRGQFDXNV4WHANCNFSM6AAAAAAYVCCTDE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

________________________________
OPPO

本电子邮件及其附件含有OPPO公司的保密信息，仅限于邮件指明的收件人（包含个人及群组）使用。禁止任何人在未经授权的情况下以任何形式使用。如果您错收了本邮件，切勿传播、分发、复制、印刷或使用本邮件之任何部分或其所载之任何内容，并请立即以电子邮件通知发件人并删除本邮件及其附件。
网络通讯固有缺陷可能导致邮件被截留、修改、丢失、破坏或包含计算机病毒等不安全情况，OPPO对此类错误或遗漏而引致之任何损失概不承担责任并保留与本邮件相关之一切权利。
除非明确说明，本邮件及其附件无意作为在任何国家或地区之要约、招揽或承诺，亦无意作为任何交易或合同之正式确认。 发件人、其所属机构或所属机构之关联机构或任何上述机构之股东、董事、高级管理人员、员工或其他任何人（以下称“发件人”或“OPPO”）不因本邮件之误送而放弃其所享之任何权利，亦不对因故意或过失使用该等信息而引发或可能引发的损失承担任何责任。
文化差异披露：因全球文化差异影响，单纯以YES\OK或其他简单词汇的回复并不构成发件人对任何交易或合同之正式确认或接受，请与发件人再次确认以获得明确书面意见。发件人不对任何受文化差异影响而导致故意或错误使用该等信息所造成的任何直接或间接损害承担责任。
This e-mail and its attachments contain confidential information from OPPO, which is intended only for the person or entity whose address is listed above. Any use of the information contained herein in any way (including, but not limited to, total or partial disclosure, reproduction, or dissemination) by persons other than the intended recipient(s) is prohibited. If you are not the intended recipient, please do not read, copy, distribute, or use this information. If you have received this transmission in error, please notify the sender immediately by reply e-mail and then delete this message.
Electronic communications may contain computer viruses or other defects inherently, may not be accurately and/or timely transmitted to other systems, or may be intercepted, modified ,delayed, deleted or interfered. OPPO shall not be liable for any damages that arise or may arise from such matter and reserves all rights in connection with the email.
Unless expressly stated, this e-mail and its attachments are provided without any warranty, acceptance or promise of any kind in any country or region, nor constitute a formal confirmation or acceptance of any transaction or contract. The sender, together with its affiliates or any shareholder, director, officer, employee or any other person of any such institution (hereinafter referred to as "sender" or "OPPO") does not waive any rights and shall not be liable for any damages that arise or may arise from the intentional or negligent use of such information.
Cultural Differences Disclosure: Due to global cultural differences, any reply with only YES\OK or other simple words does not constitute any confirmation or acceptance of any transaction or contract, please confirm with the sender again to ensure clear opinion in written form. The sender shall not be responsible for any direct or indirect damages resulting from the intentional or misuse of such information.

TahaTesser commented:
> https://api.flutter.dev/flutter/material/Scrollbar-class.html#material.Scrollbar.2

Yes, tried all channels on Dartpad again to be sure.

TahaTesser commented:
Is it a good practice to add `primary: true`?

xu-baolin commented:
Yeah, I think so.
It's easier than specifying a controller explicitly, isn't it?

________________________________
来自TT移动邮箱 / From TT mailbox


________________________________
发件人: Taha Tesser ***@***.***>
发送时间: 2023年06月29日 23:01
收件人: flutter/flutter ***@***.***>
抄送: 徐宝林(Ava) ***@***.***>; Mention ***@***.***>
主题: Re: [flutter/flutter] fix a Scrollbar example crash (PR #127925)


Is it a good practice to add primary: true?

―
Reply to this email directly, view it on GitHub<https://github.com/flutter/flutter/pull/127925#issuecomment-1613340698>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOR66GDE3QWBWEWAKR6NSMTXNWKGHANCNFSM6AAAAAAYVCCTDE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

________________________________
OPPO

本电子邮件及其附件含有OPPO公司的保密信息，仅限于邮件指明的收件人（包含个人及群组）使用。禁止任何人在未经授权的情况下以任何形式使用。如果您错收了本邮件，切勿传播、分发、复制、印刷或使用本邮件之任何部分或其所载之任何内容，并请立即以电子邮件通知发件人并删除本邮件及其附件。
网络通讯固有缺陷可能导致邮件被截留、修改、丢失、破坏或包含计算机病毒等不安全情况，OPPO对此类错误或遗漏而引致之任何损失概不承担责任并保留与本邮件相关之一切权利。
除非明确说明，本邮件及其附件无意作为在任何国家或地区之要约、招揽或承诺，亦无意作为任何交易或合同之正式确认。 发件人、其所属机构或所属机构之关联机构或任何上述机构之股东、董事、高级管理人员、员工或其他任何人（以下称“发件人”或“OPPO”）不因本邮件之误送而放弃其所享之任何权利，亦不对因故意或过失使用该等信息而引发或可能引发的损失承担任何责任。
文化差异披露：因全球文化差异影响，单纯以YES\OK或其他简单词汇的回复并不构成发件人对任何交易或合同之正式确认或接受，请与发件人再次确认以获得明确书面意见。发件人不对任何受文化差异影响而导致故意或错误使用该等信息所造成的任何直接或间接损害承担责任。
This e-mail and its attachments contain confidential information from OPPO, which is intended only for the person or entity whose address is listed above. Any use of the information contained herein in any way (including, but not limited to, total or partial disclosure, reproduction, or dissemination) by persons other than the intended recipient(s) is prohibited. If you are not the intended recipient, please do not read, copy, distribute, or use this information. If you have received this transmission in error, please notify the sender immediately by reply e-mail and then delete this message.
Electronic communications may contain computer viruses or other defects inherently, may not be accurately and/or timely transmitted to other systems, or may be intercepted, modified ,delayed, deleted or interfered. OPPO shall not be liable for any damages that arise or may arise from such matter and reserves all rights in connection with the email.
Unless expressly stated, this e-mail and its attachments are provided without any warranty, acceptance or promise of any kind in any country or region, nor constitute a formal confirmation or acceptance of any transaction or contract. The sender, together with its affiliates or any shareholder, director, officer, employee or any other person of any such institution (hereinafter referred to as "sender" or "OPPO") does not waive any rights and shall not be liable for any damages that arise or may arise from the intentional or negligent use of such information.
Cultural Differences Disclosure: Due to global cultural differences, any reply with only YES\OK or other simple words does not constitute any confirmation or acceptance of any transaction or contract, please confirm with the sender again to ensure clear opinion in written form. The sender shall not be responsible for any direct or indirect damages resulting from the intentional or misuse of such information.

TahaTesser reviewed (approved):
LGTM

TahaTesser commented:
Actually, this will need a test-exemption from Hixe to land.
However given this is causing crashes, it would be better to add example tests which check `takeException` is null.

TahaTesser commented on examples/api/lib/material/scrollbar/scrollbar.0.dart:
```diff
@@ -29,6 +29,7 @@ class ScrollbarExample extends StatelessWidget {
   Widget build(BuildContext context) {
     return Scrollbar(
       child: GridView.builder(
+        primary: true,
```
Add a test for this in `api/tests` folder

TahaTesser commented:
@Piinks 
Is `primary: true` necessary for cases like this one? https://github.com/flutter/flutter/pull/127925#issuecomment-1613340698

Piinks reviewed (commented):
> Is primary: true necessary for cases like this one?

@TahaTesser on desktop, yup. Since the api docs are often viewed on the web on desktop, this change makes sense to me. Thanks @xu-baolin!

Hixie commented:
@xu-baolin Is writing a test for this on your radar?

xu-baolin committed:
add a test

TahaTesser reviewed (approved):
LGTM

TahaTesser commented:
@xu-baolin 
BTW You'll need to remove the missing test class string from `sdk/dev/bots/check_code_samples.dart` to land this.

xu-baolin committed:
++
